### PR TITLE
Fix converting old SIMD ctors to HWI ones

### LINKED ISSUE #67038: new Vector<T> for byte, short emits vpbroadcastd thus needs an imul
```c#
public Vector<byte> Bad(byte val) => new Vector<byte>(val);
```
generates on x64 with AVX2 supported:
```asm
C.Bad(Byte)
    L0000: vzeroupper
    L0003: movzx eax, r8b
    L0007: imul eax, 0x1010101
    L000d: vmovd xmm0, eax
    L0011: vpbroadcastd ymm0, xmm0
    L0016: vmovupd [rdx], ymm0
    L001a: mov rax, rdx
    L001d: vzeroupper
    L0020: ret
```
The `imul` isn't needed when instead of `vpbroadcastd` the `vpbroadcastb` is used, as it's done by 
```c#
public Vector256<byte> Good(byte val) => Vector256.Create(val);
```
```asm
C.Good(Byte)
    L0000: vzeroupper
    L0003: movzx eax, r8b
    L0007: vmovd xmm0, eax
    L000b: vpbroadcastb ymm0, xmm0
    L0010: vmovupd [rdx], ymm0
    L0014: mov rax, rdx
    L0017: vzeroupper
    L001a: ret
```

| Type | actual | should be |
| -- | -- | -- |
| `byte`, `sbyte` | `imul` + `vpbroadcastd` | `vpbroadcastb` |
| `ushort`, `short` | `imul` + `vpbroadcastd` | `vpbroadcastw` |
| `uint`, `int` | `vpbroadcastd` | ✔️ |
| `ulong`, `long` | `vpbroadcastq` | ✔️ |

For more see [sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgDkBXfGKASzFwBuGvSasAShwB2GXtxYBhCPgAOvADY8AyjwBu/GEJGNmLSTLkwWASRl8pufkdomJ02fNsZ7jgSwAaABxIwi4MWgAW2FAqADLYwGbulqEiAMxMpAwKDADeNAyFTBnAEBDqDACCughaHCoq0BgwACYMALwAfFU1pDa4dQ1NraFFDAVFxBkAajBgGNAAPMAAns3dAELYLQAUq80MutjqAJQd3VIwAO4Ms/NL+zCdO0eno5MzcwtQpACsSMs1k8GABxMq7R6HY5nLq3L7QP5IRSwbDNF7Q96FCaFKZw+5QRa4R6bbY7IlAqGnc4MS43O7fQnE9FvbHFPHfRGMoHdMEQXbkg6vGHdekI/7ImComDMk6Y8bUMa40UEji4CJNEm7VXqqAYSnCmnXdlLbUamVypXwn7/Ram3U88E7O16oXU5WIiVS82s1mW/GEnUYTVkwP66m040EtVmoUWz74znR+2gx1Jl3Qt1Wj0KFFo2M+hUfSOLXgyYOl9NU2ER5Ulsvews4+Mcm0Vh18nYVsOw93inOSvMYguK5tLdQQKQAc2D46n3YuRtrs+nDZHkc5y/bu2X8/Xfdz0vz1AAvkA==)
I see the same on .NET 7 Preview 2.

### PR DESCRIPTION
**Pull request by aromaa**
The code path for converting SIMD ctors to HWI ones were never reached previously. We were bailing out early because the constructors return value is `void` and the existing logic assumes that the arguments take another HWI type. Correcting this also exposes few other issues that have been addresses.

When we see a method with `void` return type we assigns its argument type as the base SIMD type. This correctly detects the `Vector<T>(T value)` constructor and bails out on the other alternative constructors.

Due to depending on what Roslyn has emitted, we might have a `NEWOBJ` or a `CALL` instruction which invokes the constructor. For a `CALL` instruction we need to read the objects address from the stack and on the former case we can use the `newobjThis` to determine the local used for the assignment and we also need to push the class handle to the stack.

Fixes #67038

### CODE REVIEW

**Comment by aromaa** [comment]
Previously for the C#
```csharp
public static Vector<byte> SIMD(byte value) => new Vector<byte>(value);
```

we had the following:
```
[000013] -A--G------                         *  ASG       simd32 (copy)
[000012] ----G--N---                         +--*  BLK       simd32<32>
[000005] -----------                         |  \--*  ADDR      byref 
[000004] -------N---                         |     \--*  LCL_VAR   simd32<System.Numerics.Vector`1[System.Byte]> V03 tmp1         
[000011] -----------                         \--*  SIMD      simd32 ubyte Cast
[000010] -----------                            \--*  SIMD      simd32 int init
[000009] -----------                               \--*  MUL       int   
[000007] -----------                                  +--*  CAST      int <- int
[000000] -----------                                  |  \--*  LCL_VAR   ubyte  V01 arg0         
[000008] -----------                                  \--*  CNS_INT   int    0x1010101
```
```asm
IN0001: 000003 movzx    rax, dl
IN0002: 000006 imul     eax, eax, 0x1010101
IN0003: 00000C vmovd    xmm0, eax
IN0004: 000010 vpbroadcastd ymm0, ymm0
IN0005: 000015 vmovupd  ymmword ptr[rcx], ymm0
IN0006: 000019 mov      rax, rcx
```

and now we end up generating:
```
[000009] -A--G------                         *  ASG       simd32 (copy)
[000008] ----G--N---                         +--*  BLK       simd32<32>
[000005] -----------                         |  \--*  ADDR      byref 
[000004] -------N---                         |     \--*  LCL_VAR   simd32<System.Numerics.Vector`1[System.Byte]> V03 tmp1         
[000007] -----------                         \--*  HWINTRINSIC simd32 ubyte Create
[000000] -----------                            \--*  LCL_VAR   ubyte  V01 arg0  
```
```asm
IN0001: 000003 movzx    rax, dl
IN0002: 000006 vmovd    xmm0, eax
IN0003: 00000A vpbroadcastb ymm0, ymm0
IN0004: 00000F vmovupd  ymmword ptr[rcx], ymm0
IN0005: 000013 mov      rax, rcx
```
We are now able to take advantage of the optimizations made for HWINTRINSIC nodes.

## JIT-diff on JIT/SIMD tests:
```

Summary of Code Size diffs:
(Lower is better)

Total bytes of base: 589894
Total bytes of diff: 587413
Total bytes of delta: -2481 (-0.42 % of base)
Total relative delta: -5.44
    diff is an improvement.
    relative diff is an improvement.
```
<details>

<summary>Detail diffs</summary>

```


Top file regressions (bytes):
          28 : VectorIntEquals_ro\VectorIntEquals_ro.dasm (0,23 % of base)
          15 : Ldfld_ro\Ldfld_ro.dasm (4,19 % of base)
           5 : CreateGeneric_ro\CreateGeneric_ro.dasm (0,96 % of base)

Top file improvements (bytes):
        -309 : VectorDot_ro\VectorDot_ro.dasm (-1,87 % of base)
        -240 : VectorAdd_ro\VectorAdd_ro.dasm (-1,45 % of base)
        -232 : VectorSub_ro\VectorSub_ro.dasm (-1,45 % of base)
        -230 : VectorReturn_ro\VectorReturn_ro.dasm (-1,15 % of base)
        -210 : VectorMin_ro\VectorMin_ro.dasm (-1,25 % of base)
        -210 : VectorMax_ro\VectorMax_ro.dasm (-1,25 % of base)
        -136 : VectorDiv_ro\VectorDiv_ro.dasm (-0,78 % of base)
        -110 : SimpleSIMDProgram\SimpleSIMDProgram.dasm (-0,96 % of base)
         -86 : VectorMul_ro\VectorMul_ro.dasm (-0,38 % of base)
         -63 : VectorSqrt_ro\VectorSqrt_ro.dasm (-0,28 % of base)
         -57 : VectorArray_ro\VectorArray_ro.dasm (-0,36 % of base)
         -55 : BoxUnbox_ro\BoxUnbox_ro.dasm (-32,16 % of base)
         -53 : VectorSum_ro\VectorSum_ro.dasm (-0,39 % of base)
         -52 : VectorExp_ro\VectorExp_ro.dasm (-0,26 % of base)
         -52 : VectorGet_ro\VectorGet_ro.dasm (-0,22 % of base)
         -46 : StoreElement_ro\StoreElement_ro.dasm (-29,49 % of base)
         -45 : VectorAbs_ro\VectorAbs_ro.dasm (-0,28 % of base)
         -44 : Sums_ro\Sums_ro.dasm (-4,31 % of base)
         -42 : VectorHWAccel2_ro\VectorHWAccel2_ro.dasm (-0,31 % of base)
         -41 : SqrtGeneric_ro\SqrtGeneric_ro.dasm (-3,12 % of base)

37 total files with Code Size differences (34 improved, 3 regressed), 18 unchanged.

Top method regressions (bytes):
          28 (16,09 % of base) : VectorIntEquals_ro\VectorIntEquals_ro.dasm - VectorTest:VectorIntEquals():int
          15 (4,31 % of base) : Ldfld_ro\Ldfld_ro.dasm - VectorMathTests.Program:Main(System.String[]):int
           5 (0,97 % of base) : CreateGeneric_ro\CreateGeneric_ro.dasm - VectorMathTests.Program:Main(System.String[]):int
           4 (0,64 % of base) : VectorGet_ro\VectorGet_ro.dasm - VectorGetTest`1[Double][System.Double]:VectorGet(double,int):int
           4 (0,17 % of base) : VectorRelOp_ro\VectorRelOp_ro.dasm - VectorRelopTest`1[Double][System.Double]:VectorRelOp(double,double):int
           2 (0,34 % of base) : Ctors_ro\Ctors_ro.dasm - Program:Vector4Ctors():int

Top method improvements (bytes):
        -249 (-7,99 % of base) : VectorDot_ro\VectorDot_ro.dasm - VectorTest:Main():int
        -180 (-7,77 % of base) : VectorSub_ro\VectorSub_ro.dasm - VectorTest:Main():int
        -178 (-6,12 % of base) : VectorAdd_ro\VectorAdd_ro.dasm - VectorTest:Main():int
        -150 (-4,72 % of base) : VectorMin_ro\VectorMin_ro.dasm - VectorTest:Main():int
        -150 (-4,72 % of base) : VectorMax_ro\VectorMax_ro.dasm - VectorTest:Main():int
         -76 (-2,86 % of base) : VectorDiv_ro\VectorDiv_ro.dasm - VectorTest:Main():int
         -60 (-4,25 % of base) : VectorReturn_ro\VectorReturn_ro.dasm - VectorTest:init()
         -55 (-32,35 % of base) : BoxUnbox_ro\BoxUnbox_ro.dasm - VectorMathTests.Program:Main(System.String[]):int
         -46 (-29,68 % of base) : StoreElement_ro\StoreElement_ro.dasm - VectorTests.Program:Main(System.String[]):int
         -44 (-9,57 % of base) : Sums_ro\Sums_ro.dasm - VectorMathTests.Program:sum(System.Numerics.Vector4[]):float
         -42 (-3,22 % of base) : VectorReturn_ro\VectorReturn_ro.dasm - VectorTest:Main():int
         -41 (-3,65 % of base) : SqrtGeneric_ro\SqrtGeneric_ro.dasm - VectorMathTests.Program:Main(System.String[]):int
         -40 (-2,10 % of base) : VectorCeilFloor_ro\VectorCeilFloor_ro.dasm - VectorTest:Main():int
         -35 (-3,07 % of base) : VectorSum_ro\VectorSum_ro.dasm - VectorTest:Main():int
         -34 (-2,22 % of base) : VectorExp_ro\VectorExp_ro.dasm - VectorExpTest`1[Int16][System.Int16]:VectorExp(System.Numerics.Vector`1[Int16],short,short,short):int
         -34 (-1,83 % of base) : VectorSqrt_ro\VectorSqrt_ro.dasm - VectorSqrtTest`1[Int16][System.Int16]:VectorSqrt(short,short,short):int
         -32 (-5,57 % of base) : DivSignedUnsignedTest_ro\DivSignedUnsignedTest_ro.dasm - VectorMathTests.Program:Main(System.String[]):int
         -31 (-10,30 % of base) : AbsGeneric_ro\AbsGeneric_ro.dasm - VectorMathTests.Program:Main(System.String[]):int
         -28 (-9,82 % of base) : VectorReturn_ro\VectorReturn_ro.dasm - VectorTest:F1_v2(float):System.Numerics.Vector2
         -28 (-7,91 % of base) : VectorReturn_ro\VectorReturn_ro.dasm - VectorTest:F1_v3(float):System.Numerics.Vector3

Top method regressions (percentages):
          28 (16,09 % of base) : VectorIntEquals_ro\VectorIntEquals_ro.dasm - VectorTest:VectorIntEquals():int
          15 (4,31 % of base) : Ldfld_ro\Ldfld_ro.dasm - VectorMathTests.Program:Main(System.String[]):int
           5 (0,97 % of base) : CreateGeneric_ro\CreateGeneric_ro.dasm - VectorMathTests.Program:Main(System.String[]):int
           4 (0,64 % of base) : VectorGet_ro\VectorGet_ro.dasm - VectorGetTest`1[Double][System.Double]:VectorGet(double,int):int
           2 (0,34 % of base) : Ctors_ro\Ctors_ro.dasm - Program:Vector4Ctors():int
           4 (0,17 % of base) : VectorRelOp_ro\VectorRelOp_ro.dasm - VectorRelopTest`1[Double][System.Double]:VectorRelOp(double,double):int

Top method improvements (percentages):
         -55 (-32,35 % of base) : BoxUnbox_ro\BoxUnbox_ro.dasm - VectorMathTests.Program:Main(System.String[]):int
         -46 (-29,68 % of base) : StoreElement_ro\StoreElement_ro.dasm - VectorTests.Program:Main(System.String[]):int
          -5 (-14,71 % of base) : VectorReturn_ro\VectorReturn_ro.dasm - VectorTest:GetVector3One():System.Numerics.Vector3
         -12 (-13,64 % of base) : VectorDot_ro\VectorDot_ro.dasm - Vector2Test:VectorDot(float,float,float):int
         -12 (-13,64 % of base) : VectorDot_ro\VectorDot_ro.dasm - Vector3Test:VectorDot(float,float,float):int
         -22 (-13,41 % of base) : SimpleSIMDProgram\SimpleSIMDProgram.dasm - FuncEvalTest:RunTests()
         -24 (-12,37 % of base) : VectorAdd_ro\VectorAdd_ro.dasm - VectorAddTest`1[Int16][System.Int16]:VectorAdd(short,short,short):int
         -24 (-12,37 % of base) : VectorHWAccel_ro\VectorHWAccel_ro.dasm - VectorHWAccelTest`1[Int16][System.Int16]:VectorHWAccel(short,short,short):int
         -24 (-12,37 % of base) : VectorHWAccel2_ro\VectorHWAccel2_ro.dasm - VectorHWAccelTest`1[Int16][System.Int16]:VectorHWAccel2(short,short,short):int
         -24 (-12,37 % of base) : VectorMax_ro\VectorMax_ro.dasm - VectorMaxTest`1[Int16][System.Int16]:VectorMax(short,short,short):int
         -24 (-12,37 % of base) : VectorMin_ro\VectorMin_ro.dasm - VectorMinTest`1[Int16][System.Int16]:VectorMin(short,short,short):int
         -24 (-12,37 % of base) : VectorSub_ro\VectorSub_ro.dasm - VectorSubTest`1[Int16][System.Int16]:VectorSub(short,short,short):int
         -12 (-11,43 % of base) : VectorSum_ro\VectorSum_ro.dasm - VectorSumTest`1[Int16][System.Int16]:VectorSum(short,short):int
         -24 (-10,86 % of base) : VectorDot_ro\VectorDot_ro.dasm - VectorDotTest`1[Int16][System.Int16]:VectorDot(short,short,short):int
         -31 (-10,30 % of base) : AbsGeneric_ro\AbsGeneric_ro.dasm - VectorMathTests.Program:Main(System.String[]):int
         -28 (-9,82 % of base) : VectorReturn_ro\VectorReturn_ro.dasm - VectorTest:F1_v2(float):System.Numerics.Vector2
         -44 (-9,57 % of base) : Sums_ro\Sums_ro.dasm - VectorMathTests.Program:sum(System.Numerics.Vector4[]):float
         -12 (-9,16 % of base) : VectorMax_ro\VectorMax_ro.dasm - Vector2Test:VectorMax(float,float,float):int
         -12 (-9,16 % of base) : VectorMin_ro\VectorMin_ro.dasm - Vector2Test:VectorMin(float,float,float):int
         -12 (-9,16 % of base) : VectorMul_ro\VectorMul_ro.dasm - Vector2Test:VectorMul(float,float,float):int

108 total methods with Code Size differences (102 improved, 6 regressed), 2068 unchanged.

```

</details>






## Example of improvement
```diff
@@ -7,63 +7,56 @@
 ; Final local variable assignments
 ;
 ;* V00 arg0         [V00    ] (  0,  0   )     ref  ->  zero-ref    class-hnd single-def
-;  V01 loc0         [V01,T01] (  4,  3.50)  simd32  ->  mm6         ld-addr-op
+;* V01 loc0         [V01,T01] (  0,  0   )  simd32  ->  zero-ref    ld-addr-op
 ;  V02 loc1         [V02,T00] (  6,  5.50)     ref  ->  rax         class-hnd exact single-def
 ;  V03 OutArgs      [V03    ] (  1,  1   )  lclBlk (32) [rsp+00H]   "OutgoingArgSpace"
 ;
-; Lcl frame size = 72
+; Lcl frame size = 40
 
 G_M32669_IG01:
-       sub      rsp, 72
+       sub      rsp, 40
        vzeroupper 
-       vmovaps  qword ptr [rsp+30H], xmm6
-       vmovaps  qword ptr [rsp+20H], xmm7
-						;; size=19 bbWeight=1    PerfScore 5.25
+						;; size=7 bbWeight=1    PerfScore 1.25
 G_M32669_IG02:
-       mov      eax, 1
-       vmovd    xmm6, eax
-       vpbroadcastd ymm6, ymm6
        mov      rcx, 0xD1FFAB1E      ; System.Numerics.Vector`1[System.Int32][]
        mov      edx, 10
-       vextractf128 xmm7, ymm6, 1
        call     CORINFO_HELP_NEWARR_1_VC
-       vinsertf128 ymm6, ymm6, xmm7, 1
-       vmovupd  ymmword ptr[rax+16], ymm6
-       vmovupd  ymmword ptr[rax+80], ymm6
+       vmovupd  ymm0, ymmword ptr[reloc @RWD00]
+       vmovupd  ymmword ptr[rax+16], ymm0
+       vmovupd  ymmword ptr[rax+80], ymm0
        vmovupd  ymm0, ymmword ptr[rax+16]
        vpcmpeqd ymm0, ymm0, ymmword ptr[rax+48]
        vpmovmskb edx, ymm0
        cmp      edx, -1
        je       SHORT G_M32669_IG04
-						;; size=75 bbWeight=1    PerfScore 26.00
+						;; size=57 bbWeight=1    PerfScore 21.75
 G_M32669_IG03:
-       vpcmpeqd ymm0, ymm6, ymmword ptr[rax+80]
+       vmovupd  ymm0, ymmword ptr[rax+80]
+       vpcmpeqd ymm0, ymm0, ymmword ptr[reloc @RWD00]
        vpmovmskb eax, ymm0
        cmp      eax, -1
        je       SHORT G_M32669_IG06
-						;; size=14 bbWeight=0.50 PerfScore 3.62
+						;; size=22 bbWeight=0.50 PerfScore 5.62
 G_M32669_IG04:
        xor      eax, eax
 						;; size=2 bbWeight=0.50 PerfScore 0.12
 G_M32669_IG05:
-       vmovaps  xmm6, qword ptr [rsp+30H]
-       vmovaps  xmm7, qword ptr [rsp+20H]
        vzeroupper 
-       add      rsp, 72
+       add      rsp, 40
        ret      
-						;; size=20 bbWeight=0.50 PerfScore 5.12
+						;; size=8 bbWeight=0.50 PerfScore 1.12
 G_M32669_IG06:
        mov      eax, 100
 						;; size=5 bbWeight=0.50 PerfScore 0.12
 G_M32669_IG07:
-       vmovaps  xmm6, qword ptr [rsp+30H]
-       vmovaps  xmm7, qword ptr [rsp+20H]
        vzeroupper 
-       add      rsp, 72
+       add      rsp, 40
        ret      
-						;; size=20 bbWeight=0.50 PerfScore 5.12
+						;; size=8 bbWeight=0.50 PerfScore 1.12
+RWD00  	dq	0000000100000001h, 0000000100000001h, 0000000100000001h, 0000000100000001h
 
-; Total bytes of code 155, prolog size 19, PerfScore 62.28, instruction count 35, allocated bytes for code 169 (MethodHash=5c358062) for method VectorTests.Program:Main(System.String[]):int
+
+; Total bytes of code 109, prolog size 7, PerfScore 42.93, instruction count 26, allocated bytes for code 118 (MethodHash=5c358062) for method VectorTests.Program:Main(System.String[]):int
```

File: src/coreclr/jit/hwintrinsic.cpp [hunk 1]
```diff
@@ -497,6 +497,14 @@ GenTree* Compiler::getArgForHWIntrinsic(var_types            argType,
             impPushOnStack(gtNewLclvNode(tmp, lvaGetRealType(tmp)), verMakeTypeInfo(argClass).NormaliseForStack());
         }
     }
+    else if (argType == TYP_REF)
```
Comment (tannergooding) [inline src/coreclr/jit/hwintrinsic.cpp:500]: This is basically duplicating everything from the `else` path but with a slightly different assert.

Can we just merge the two paths?

File: src/coreclr/jit/simdashwintrinsic.cpp [hunk 2]
```diff
@@ -918,16 +925,24 @@ GenTree* Compiler::impSimdAsHWIntrinsicSpecial(NamedIntrinsic       intrinsic,
             if (SimdAsHWIntrinsicInfo::SpillSideEffectsOp1(intrinsic))
             {
                 impSpillSideEffect(true, verCurrentState.esStackDepth -
-                                             2 DEBUGARG("Spilling op1 side effects for SimdAsHWIntrinsic"));
+                                             (newobjThis == nullptr ? 2 : 1)DEBUGARG(
```
Comment (tannergooding) [inline src/coreclr/jit/simdashwintrinsic.cpp:928]: Why does this need `2` vs `1`? Whether `newobjThis` is there or not, we should still have exactly two operands on the stack that need to be popped and consumed here.

File: src/coreclr/jit/simdashwintrinsic.cpp [hunk 3]
```diff
@@ -918,16 +925,24 @@ GenTree* Compiler::impSimdAsHWIntrinsicSpecial(NamedIntrinsic       intrinsic,
...
-            op1 = getArgForHWIntrinsic(argType, argClass, isInstanceMethod, newobjThis);
+            bool implicitConstructor = isInstanceMethod && (newobjThis == nullptr) && retType == TYP_VOID;
+
+            argType = implicitConstructor
+                          ? TYP_REF
+                          : isInstanceMethod
+                                ? simdType
+                                : JITtype2varType(strip(info.compCompHnd->getArgType(sig, argList, &argClass)));
+
+            op1 = getArgForHWIntrinsic(argType, (newobjThis != nullptr) ? clsHnd : argClass, isInstanceMethod,
```
Comment (tannergooding) [inline src/coreclr/jit/simdashwintrinsic.cpp:944]: When would `clsHnd` and `argClass` be different?

File: src/coreclr/jit/hwintrinsic.cpp (hunk 1 above)
Comment (aromaa) [reply src/coreclr/jit/hwintrinsic.cpp:500]: Are you suggesting to merge the asserts or making them more relaxed?

File: src/coreclr/jit/simdashwintrinsic.cpp [hunk 4]
```diff
@@ -205,8 +205,15 @@ GenTree* Compiler::impSimdAsHWIntrinsic(NamedIntrinsic        intrinsic,
     }
     else if (numArgs != 0)
     {
-        argClass        = info.compCompHnd->getArgClass(sig, sig->args);
-        simdBaseJitType = getBaseJitTypeAndSizeOfSIMDType(argClass, &simdSize);
+        if (retType == TYP_VOID)
+        {
+            simdBaseJitType = strip(info.compCompHnd->getArgType(sig, sig->args, &argClass));
```
[the remark this answers is not in the document]
Comment (aromaa) [reply src/coreclr/jit/simdashwintrinsic.cpp:210]: Right, we don't actually have any methods that would match those right now. Checking just for instance one would be more strict and make this  `sig->hasThis() && (retType == TYP_VOID)`

File: src/coreclr/jit/simdashwintrinsic.cpp (hunk 2 above)
Comment (aromaa) [reply src/coreclr/jit/simdashwintrinsic.cpp:928]: `newobjThis` is used when the `NEWOBJ` instruction is used. We don't have the address in the stack here.

File: src/coreclr/jit/hwintrinsic.cpp (hunk 1 above)
Comment (tannergooding) [reply src/coreclr/jit/hwintrinsic.cpp:500]: Likely merging them.

File: src/coreclr/jit/simdashwintrinsic.cpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (tannergooding) [reply src/coreclr/jit/simdashwintrinsic.cpp:210]: Right, and also less error prone, so I think its desirable.

File: src/coreclr/jit/simdashwintrinsic.cpp (hunk 3 above)
Comment (aromaa) [reply src/coreclr/jit/simdashwintrinsic.cpp:944]: The `argType`  gets re-assigned few lines above. This ends up being `T` instead of `Vector<T>` here.

**Comment by JulieLeeMSFT** [comment]
@aromaa, we are at the end of .NET 7 release cycle. Once we are less busy, @kunalspathak will get back to review your PR. Thanks for your contribution.

**Comment by kunalspathak** [comment]
I will review it this week.

**Review by kunalspathak — APPROVED** [review]

**Pull request merged**